### PR TITLE
Add metrics test

### PR DESCRIPTION
**Pull request by mrwwheat7**

### CODE REVIEW

File: scripts/test.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import requests
```
Comment (pzheng1025) [inline scripts/test.py:1]: give a more specific name instead of test.py

Comment (mrwwheat7) [reply scripts/test.py:1]: Sure, this one is just to test the github workflows...When it is ready for review, will change its name and add more comments

File: scripts/datenlord_metrics_test.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+set -o xtrace
+
+# Need to wait for some time before metrics are collected by Promethues
+promethues_wait_time=5 
+
+kubectl apply -f $DATENLORD_METRICS_TEST
+kubectl wait --for=condition=Ready pod mysql-datenlord-test --timeout=60s
+kubectl exec -i mysql-datenlord-test -- bash -c "echo test > /usr/share/nginx/html/testfile"
+kubectl exec -i mysql-datenlord-test -- bash -c "cat /usr/share/nginx/html/testfile"
+sleep $promethues_wait_time
```
Comment (pwang7) [inline scripts/datenlord_metrics_test.sh:14]: use `PROMETHUES_WAIT_TIME` for shell env var

File: scripts/datenlord_metrics_test.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+NODE_IP=`kubectl get nodes -A -o wide | awk 'FNR == 2 {print $6}'`
+curl -s "http://${NODE_IP}:30000/api/v1/query?query=datenlord_cache_hits" | python -c "import sys, json; assert json.load(sys.stdin)['data']['result'][0]['value'][1] == '1'"
+curl -s "http://${NODE_IP}:30000/api/v1/query?query=datenlord_cache_misses" | python -c "import sys, json; assert json.load(sys.stdin)['data']['result'][0]['value'][1] == '1'"
```
Comment (pwang7) [inline scripts/datenlord_metrics_test.sh:17]: add one empty line below

**Review by pwang7 — CHANGES_REQUESTED** [review]

File: scripts/datenlord_metrics_test.sh (hunk 2 above)
Comment (mrwwheat7) [reply scripts/datenlord_metrics_test.sh]: Done

File: scripts/datenlord_metrics_test.sh (hunk 3 above)
Comment (mrwwheat7) [reply scripts/datenlord_metrics_test.sh]: Done

**Review by pwang7 — CHANGES_REQUESTED** [review]

File: scripts/datenlord_monitor_test.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+set -o errexit
+set -o nounset
+set -o xtrace
+
+sudo chmod +x scripts/datenlord-monitor-deploy.sh
+cat scripts/alertmanager_test_alert.yaml >> scripts/alertmanager_alerts.yaml
+docker pull quay.io/coreos/kube-state-metrics:v1.8.0
+docker pull prom/alertmanager:v0.19.0
+docker pull prom/prometheus
```
Comment (mrwwheat7) [inline scripts/datenlord_monitor_test.sh:11]: Use specific version

File: .github/workflows/cron.yml [hunk 5]
```diff
@@ -267,6 +263,13 @@ jobs:
           sleep 60
           kubectl get pods -A -o wide
           # sudo netstat -lntp && ls -lsh
+      - name: Datenlord metric test
+        run: |
+          sh scripts/datenlord_metrics_test.sh
+      - name: Datenlord logging test
+        run: |
+          NODE_IP=`kubectl get nodes -A -o wide | awk 'FNR == 2 {print $6}'`
```
Comment (mrwwheat7) [inline .github/workflows/cron.yml:271]: Merge this to metrics test

File: .github/workflows/cron.yml [hunk 6]
```diff
@@ -215,12 +216,7 @@ jobs:
...
-          sudo chmod 755 scripts/datenlord-monitor-deploy.sh
-          cat scripts/alertmanager_test_alert.yaml >> scripts/alertmanager_alerts.yaml
-          sh scripts/datenlord-monitor-deploy.sh deploy
-          NODE_IP=`kubectl get nodes -A -o wide | awk 'FNR == 2 {print $6}'`
-          FOUND_PATH=`curl --silent $NODE_IP:30000 | grep Found`
-          test -n "$FOUND_PATH" || (echo "FAILED TO FIND PROMETHEUS SERVICE" && /bin/false)
+          sh scripts/datenlord_monitor_test.sh
       - name: Datenlord Alerting Test
         run: |
           sleep 60
```
Comment (mrwwheat7) [inline .github/workflows/cron.yml:222]: Merge this to monitor test

File: .github/workflows/cron.yml [hunk 7]
```diff
@@ -15,6 +15,7 @@ env:
   DATENLORD_IMAGE: datenlord/datenlord:e2e_test
   DATENLORD_LOGGING: scripts/datenlord-logging.yaml
   DATENLORD_LOGGING_NAMESPACE: datenlord-logging
+  DATENLORD_METRICS_TEST: scripts/datenlord-metrics-test.yaml
```
Comment (mrwwheat7) [inline .github/workflows/cron.yml:18]: Add env variable here

File: scripts/datenlord-monitor-deploy.sh [hunk 8]
```diff
@@ -24,15 +24,11 @@ else
...
-    docker pull quay.io/coreos/kube-state-metrics:v1.8.0
-    docker pull prom/alertmanager:v0.19.0
-    docker pull prom/prometheus
-    docker pull grafana/grafana:latest
     kubectl apply -f scripts/alertmanager_alerts.yaml
     kubectl apply -f scripts/datenlord-logging.yaml
     kubectl apply -f scripts/datenlord-monitor.yaml
-    kubectl wait --for=condition=Ready pod -l app=prometheus-server -n datenlord-monitoring --timeout=60s
-    kubectl wait --for=condition=Ready pod -l app=grafana -n datenlord-monitoring --timeout=60s
+    kubectl wait --for=condition=Ready pod -l app=prometheus-server -n datenlord-monitoring --timeout=120s
```
Comment (mrwwheat7) [inline scripts/datenlord-monitor-deploy.sh:30]: Change timeout to 5s

**Review by mrwwheat7 — COMMENTED** [review]
Change image version in yaml as env variable

**Review by pzheng1025 — APPROVED** [review]

**Review by pwang7 — APPROVED** [review]

**Pull request merged**